### ISSUE TITLE
Pow `^` does not compile when using `unchained`

### ISSUE THREAD

**Issue by arkanoid87** [SciNim/Measuremancer#7]
```nim
import unchained
import measuremancer

let
    a = 2.1.m ± 0.5.m
    b = 1.2.m ± 0.3.m

echo a**b
```

```nim
measuremancer.nim(312, 17) Error: type mismatch: got <Meter, Meter>
but expected one of:
func pow(x, y: float32): float32
  first type mismatch at position: 1
  required type for x: float32
  but expression 'a.val' is of type: Meter
func pow(x, y: float64): float64
  first type mismatch at position: 1
  required type for x: float64
  but expression 'a.val' is of type: Meter
```

**Comment by arkanoid87** [comment]
Speaking of power operator, I think there should also be this version

```nim
proc `**`*[T: FloatLike](m: FloatLike, p: Measurement[T]): Measurement[T] =`
```

**Comment by arkanoid87** [comment]
actually, it doesn't work even without units

```nim
import unchained
import measuremancer

let
    u = 2.1 ± 0.5
    z = 1.2 ± 0.3

echo u**z
```

```nim
measuremancer.nim(315, 31) Error: type mismatch: got <float>
but expected one of:
func log[T: SomeFloat](x, base: T): T
  first type mismatch at position: 2
  missing parameter: base
```
math.log wants second `base` argument
https://nim-lang.org/docs/math.html#log%2CT%2CT

**Comment by arkanoid87** [comment]
I need to add that the more I read the code (to hopefully grasp how to create a PR) the more I understand the complexity of `unchained` integration.

**Comment by arkanoid87** [comment]
I've been following the white rabbit until I've decided to properly turn it into tests
https://github.com/SciNim/Measuremancer/pull/8

**Comment by arkanoid87** [comment]
I'm trying some fixes in my fork https://github.com/SciNim/Measuremancer/commit/a5f06897743ec67ed280ba25ee417a0e99c8cc0c#diff-c3e2bf281e6b65db974cf236911d68cb47f1f0a14bb0029e5354e573fc2da0eb

I fixed power & exponentials when not using unchained (comparing results with vs python `uncertainties`), but how to deal with float exponents?

Python Pint handles them at runtime using strings
```python
from pint import UnitRegistry

ureg = UnitRegistry()

distance = 24.0 * ureg.meter
print(repr(distance ** 12.5))
print(repr(distance ** distance))
```

Unchained hates this to live at compile life
```python
<Quantity(1.78912433e+17, 'meter ** 12.5')>
Traceback (most recent call last):
pint.errors.DimensionalityError: Cannot convert from 'meter' to 'dimensionless'
```

```nim
import unchained

echo pow(5.m, 4.2)
```
```nim
Error: type mismatch: got <Meter, float64>
but expected one of:
func pow(x, y: float32): float32
  first type mismatch at position: 1
  required type for x: float32
  but expression 'm(5)' is of type: Meter
func pow(x, y: float64): float64
  first type mismatch at position: 1
  required type for x: float64
  but expression 'm(5)' is of type: Meter

expression: pow(m(5), 4.2)
```

should I consider units as base or exponent of power/exponential a compile error?

**Comment by Vindaar** [comment]
Float based exponents simply don't make sense with units attached! What's a 5.kg^{2.3} ? The same is true for trigonometric functions. Their argument must be `UnitLess` (or an equivalent like `Rad`).

Haven't had time to properly look yet, sorry.

**Comment by arkanoid87** [comment]
No need to be sorry, you wrapped up an amazing system. I just need some time to wrap my head around all of this

**Comment by Vindaar** [comment]
Closing this, as it should be addressed in version `v0.2.2`. Feel free to reopen if I missed something.
